loopflow-computation/src/main/java/com/farao_community/farao/loopflow_computation/LoopFlowComputation.java

```diff
@@ -6,179 +6,102 @@
-        sensiResults.getSensitivityValues().forEach(sensitivityValue -> addSensitivityValue(sensitivityValue, crac, ptdfs));
-        return ptdfs;
-    }
+        SystematicSensitivityResult ptdfsAndRefFlows = systematicSensitivityInterface.run(network, Unit.MEGAWATT);
 
-    private List<SensitivityFactor> generateSensitivityFactorsProvider(Network network, Set<Cnec> cnecs, GlskProvider glskProvider) {
-        List<SensitivityFactor> factors = new ArrayList<>();
-        Map<String, LinearGlsk> mapCountryLinearGlsk = glskProvider.getAllGlsk(network);
-        cnecs.forEach(cnec -> mapCountryLinearGlsk.values().stream()
-                .map(linearGlsk -> new BranchFlowPerLinearGlsk(new BranchFlow(cnec.getId(), cnec.getName(), cnec.getNetworkElement().getId()), linearGlsk))
-                .forEach(factors::add));
-        return factors;
+        return buildLoopFlowsFromReferenceFlowAndPtdf(ptdfsAndRefFlows, network);
     }
 
-    private void addSensitivityValue(SensitivityValue sensitivityValue, Crac crac, Map<Cnec, Map<Country, Double>> ptdfs) {
-        String cnecId = sensitivityValue.getFactor().getFunction().getId();
-        Cnec cnec = crac.getCnec(cnecId);
-        String glskId = sensitivityValue.getFactor().getVariable().getId();
-        double ptdfValue = sensitivityValue.getValue();
-        if (!ptdfs.containsKey(cnec)) {
-            ptdfs.put(cnec, new HashMap<>());
-        }
-        ptdfs.get(cnec).put(glskIdToCountry(glskId), ptdfValue);
-    }
+    public LoopFlowResult buildLoopFlowsFromReferenceFlowAndPtdf(SystematicSensitivityResult alreadyCalculatedPtdfAndFlows, Network network) {
+        return buildLoopFlowsFromReferenceFlowAndPtdf(alreadyCalculatedPtdfAndFlows, alreadyCalculatedPtdfAndFlows, network);
 
-    private Country glskIdToCountry(String glskId) {
-        if (glskId.length() < EICode.LENGTH) {
-            throw new IllegalArgumentException(String.format("GlskId [%s] should starts with an EI Code", glskId));
-        }
-        EICode eiCode = new EICode(glskId.substring(0, EICode.LENGTH));
-        return eiCode.getCountry();
     }
 
-    public Map<Cnec, Double> computeRefFlowOnCurrentNetwork(Network network) {
-        // we need this separate load flow to get reference flow on cnec.
-        // because reference flow from sensi is not yet fully implemented in powsybl
-        Map<Cnec, Double> cnecFlowMap = new HashMap<>();
-        String initialVariantId = network.getVariantManager().getWorkingVariantId();
-        // 1. pre
-        LoadFlowResult loadFlowResult = LoadFlowService.runLoadFlow(network, initialVariantId);
-        if (loadFlowResult.isOk()) {
-            buildFlowFromNetwork(network, crac, cnecFlowMap);
-        }
-        return cnecFlowMap;
-    }
+    public LoopFlowResult buildLoopFlowsFromReferenceFlowAndPtdf(SystematicSensitivityResult alreadyCalculatedFlows, SystematicSensitivityResult alreadyCalculatedPtdfs, Network network) {
```

Comment:
Only one input is needed (cf previous remark)

Reply:
method removed